### PR TITLE
aptos-client: support get table item at a specific version

### PR DESCRIPTION
**Pull request by nanne007**
### Description

client api supports get table item(json or bcs) at some specific version.

### Test Plan
No test needed.

### CODE REVIEW

File: crates/aptos-rest-client/src/lib.rs [hunk 1]
```diff
@@ -1274,6 +1292,24 @@ impl Client {
         let response = self.post_bcs(url, data).await?;
         Ok(response.and_then(|inner| bcs::from_bytes(&inner))?)
     }
+    pub async fn get_table_item_bcs_at_version<K: Serialize, T: DeserializeOwned>(
```
Comment (davidiw) [inline crates/aptos-rest-client/src/lib.rs:1295]: nit on space above
```suggestion

    pub async fn get_table_item_bcs_at_version<K: Serialize, T: DeserializeOwned>(
```

File: crates/aptos-rest-client/src/lib.rs [hunk 2]
```diff
@@ -1256,6 +1256,24 @@ impl Client {
         let response = self.inner.post(url).json(&data).send().await?;
         self.json(response).await
     }
+    pub async fn get_table_item_at_version<K: Serialize>(
```
Comment (davidiw) [inline crates/aptos-rest-client/src/lib.rs:1259]:
```suggestion

    pub async fn get_table_item_at_version<K: Serialize>(
```

**Review by davidiw — APPROVED** [review]

**Review by heliuchuan — APPROVED** [review]

File: crates/aptos-rest-client/src/lib.rs [hunk 3]
```diff
@@ -1257,6 +1257,25 @@ impl Client {
...
 
+    pub async fn get_table_item_at_version<K: Serialize>(
+        &self,
+        table_handle: AccountAddress,
+        key_type: &str,
+        value_type: &str,
+        key: K,
+        version: u64,
+    ) -> AptosResult<Response<Value>> {
+        let url = self.build_path(&format!("tables/{}/item?ledger_version={}", table_handle,version))?;
```
Comment (davidiw) [inline crates/aptos-rest-client/src/lib.rs:1268]:
```suggestion
        let url = self.build_path(&format!(
            "tables/{}/item?ledger_version={}",
            table_handle, version
        ))?;
```

File: crates/aptos-rest-client/src/lib.rs [hunk 4]
```diff
@@ -1275,6 +1294,25 @@ impl Client {
...
 
+    pub async fn get_table_item_bcs_at_version<K: Serialize, T: DeserializeOwned>(
+        &self,
+        table_handle: AccountAddress,
+        key_type: &str,
+        value_type: &str,
+        key: K,
+        version: u64,
+    ) -> AptosResult<Response<T>> {
+        let url = self.build_path(&format!("tables/{}/item?ledger_version={}", table_handle, version))?;
```
Comment (davidiw) [inline crates/aptos-rest-client/src/lib.rs:1305]:
```suggestion
        let url = self.build_path(&format!(
            "tables/{}/item?ledger_version={}",
            table_handle, version
        ))?;
```

**Review by gregnazario — APPROVED** [review]

**Pull request merged**